addons/product/populate/product.py

```diff
@@ -16,21 +20,53 @@ def _populate_factories(self):
         def get_name(values=None, counter=0, complete=False, **kwargs):
             return "%s_%s_%s" % ("product_category", int(complete), counter)
 
-        # quid of parent_id ???
-
         return [("name", populate.compute(get_name))]
 
+    def _populate(self, size):
+        categories = super()._populate(size)
+        # set parent_ids
+        self._populate_set_parents(categories, size)
+        return categories
+
+    def _populate_set_parents(self, categories, size):
+        _logger.info('Setting parent categories')
+        parent_len = int(self._populate_sizes[size] / 5) # ~20% of categories may have children
+        parents = categories[::parent_len]
```

Comment:
I'm not a big fan of this solution because depending on the size, the first records won't be the same, and the first records represents the corner cases of a model. In a low populate, some of the corner case will have parent, other not. On a large populate, all corner case will be a parent category.  

A solution would be to randomly define if a category has parent or not on the fly. 

I'm not sure my version was good, because thinking about it the number of child may change depdending on the population size. But to be ideal, the choice of having a parent, and what is the parent should be seeded  by the position in the list, not the elements of the list.

Reply:
I've replaced it by a random choice of 1/16 of the categories as parents.
This way, we do not have only the x first % of the generated records as parents.